From ace99853b03bae63e1b06c2e0ec8521b755c22e9 Mon Sep 17 00:00:00 2001
From: Bo Huang <bo.b.huang@intel.com>
Date: Tue, 14 Aug 2012 17:21:59 +0800
Subject: MtpServer.cpp: ANR occurs after format one non-vfat SD card during transfer file to DUT.

BZ: 48828

Mutex is not recursive. Insert/remove storage trigger MTP server to
invoke Mutex multiple time in same thread when MTP transfer file to DUT.

Category: AOSP Improvement
Domain: USB MTP-PTP
Origin: Internal
Upstream-Candidate: No, proprietary

Change-Id: I35be71c705493b807edaae7015e4ce1542484eaf
Orig-Change-Id: Ib94a6d15e18811a8adaa0c5a158ce969f6b25a2d
Signed-off-by: Bo Huang <bo.b.huang@intel.com>
Reviewed-on: http://android.intel.com:8080/76836
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Meng, Zhe <zhe.meng@intel.com>
Tested-by: Meng, Zhe <zhe.meng@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 media/mtp/MtpServer.cpp |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/media/mtp/MtpServer.cpp b/media/mtp/MtpServer.cpp
index 3f8e4a7..d5f76ae 100644
--- a/media/mtp/MtpServer.cpp
+++ b/media/mtp/MtpServer.cpp
@@ -115,15 +115,11 @@ MtpServer::~MtpServer() {
 }
 
 void MtpServer::addStorage(MtpStorage* storage) {
-    Mutex::Autolock autoLock(mMutex);
-
     mStorages.push(storage);
     sendStoreAdded(storage->getStorageID());
 }
 
 void MtpServer::removeStorage(MtpStorage* storage) {
-    Mutex::Autolock autoLock(mMutex);
-
     for (int i = 0; i < mStorages.size(); i++) {
         if (mStorages[i] == storage) {
             mStorages.removeAt(i);
-- 
1.7.4.1

